src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -91,6 +91,35 @@ pub fn update_lockfile(manifest_path: &Path,
                                                   Method::Everything,
                                                   Some(&previous_resolve),
                                                   Some(&to_avoid)));
+
+    // Summarize what is changing for the user.
+    let print_change = |status: &str, msg: String| {
+        opts.config.shell().status(status, msg)
+    };
+    for (removed, added) in compare_dependency_graphs(&previous_resolve, &resolve) {
+        if removed.len() == 1 && added.len() == 1 {
+            if removed[0].source_id().is_git() {
+                try!(print_change("Updating", format!("{} #{} -> #{}",
+                    removed[0].name(),
```

Comment:
Hm yeah it can get long sometimes, but I think it's better than not printing it because there's real contextual information (e.g. `foo` from a git repo is different than `foo` from crates.io is different than `foo` from a path source)